frontend/catalyst/qfunc.py

```diff
@@ -267,12 +269,23 @@ def wrap_single_shot_qnode(*_):
         results = catalyst.vmap(wrap_single_shot_qnode)(arg_vmap)
         if isinstance(results[0], tuple) and len(results) == 1:
             results = results[0]
-
-        out = parse_native_mid_circuit_measurements(
-            cpy_tape, aux_tapes, results, postselect_mode="pad-invalid-samples"
-        )
-        if len(cpy_tape.measurements) == 1:
-            out = (out,)
+        has_mcm = any(isinstance(op, MidCircuitMeasure) for op in cpy_tape.operations)
+        out = list(results)
+        if has_mcm:
+            out = parse_native_mid_circuit_measurements(
+                cpy_tape, aux_tapes, results, postselect_mode="pad-invalid-samples"
+            )
+            if len(cpy_tape.measurements) == 1:
+                out = (out,)
+        else:
+            m_count = 0
+            for m in cpy_tape.measurements:
+                is_valid = jnp.array([True] * len(out[m_count]))
```

Comment:
What does this `is_valid` do?

Reply:
Oh right, this is about post-selection and filtering invalid samples I presume
```suggestion
                # Without MCMs and postselection, all samples are valid for use in MP computation.
                is_valid = jnp.array([True] * len(out[m_count]))
```